src/internal/m365/collection/drive/item.go

```diff
@@ -65,14 +65,15 @@ func downloadItem(
 		url = str.FirstIn(item.GetAdditionalData(), downloadURLKeys...)
 	}
 
-	reader, err = downloadFile(ctx, getter, url)
+	reader, err = downloadFile(ctx, getter, url, ptr.Val(item.GetSize()) < largeFileDownloadLimit)
```

Comment:
Can we add a regression test with a mocked getter or similar which tests expected behavior i.e.
1. Small files( < `largeFileDownloadLimit`)  are downloaded using the download url. Large files use `/content` endpoint. 
2. Auth flag is specified for large files, and not for small files. 

Otherwise, it'd be hard to detect regressions until we hit them in prod, since we don't test with such large files typically.

Reply:
not a bad idea.  added that.